### PR TITLE
PEDS-580_add_explorer_fields 

### PR DESCRIPTION
**Pull request by dvenckusuchgo**
change 'search' code files and references to 'filterset'
handle explorer_id in all crud functions

### CODE REVIEW

File: amanuensis/__init__.py [hunk 1]
```diff
@@ -100,8 +100,7 @@ def app_register_blueprints(app):
     # google_blueprint = amanuensis.blueprints.google.make_google_blueprint()
     # app.register_blueprint(google_blueprint, url_prefix="/google")
 
-
-    app.register_blueprint(amanuensis.blueprints.search.blueprint, url_prefix="/filter-set")
+    app.register_blueprint(amanuensis.blueprints.filterset.blueprint, url_prefix="/filter-set")
```
Comment (bobaekang) [inline amanuensis/__init__.py:103]: Shouldn't `url_prefix` value here be `"/filter-sets"` instead?

File: amanuensis/blueprints/filterset.py [hunk 2]
```diff
@@ -95,6 +124,9 @@ def delete_search(search_id):
             "Unable to load or find the user, check your token"
         )
 
-    response = flask.jsonify(delete(logged_user_id, search_id))
+    # get the explorer_id from the querystring
+    explorer_id = flask.request.args.get('explorer', default=1, type=int)
```
Comment (bobaekang) [inline amanuensis/blueprints/filterset.py:128]: I've been going back and forth on this, but now I think it's better to rename the search param to `explorerId` to match the variable name. In my view, `explorerId=1` is simply more explicit than `explorer=1`.

File: amanuensis/resources/userdatamodel/userdatamodel_filterset.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+    #TODO check that at least one has changed
+    result = current_session.query(Search).filter(
+        Search.id == filter_set_id, 
+        Search.filter_source_internal_id == explorer_id,
+        Search.filter_source == FilterSourceType.explorer,
+        Search.user_id == logged_user_id
+    ).update(data)
+    if  result > 0:
+        return  {"code": 200, "updated": filter_set_id, "explorer_id": explorer_id}
```
Comment (bobaekang) [inline amanuensis/resources/userdatamodel/userdatamodel_filterset.py:74]: If I'm correct, `filter_set_id` is a string value here, originally extracted from URL in PUT request and passed down as a param. However, I think it makes more sense to return the actual int value in the response body. Can we make that change? It seems that we can use `result.id` here.

File: amanuensis/resources/userdatamodel/userdatamodel_filterset.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    data = {}
+    if name:
+        data['name'] = name
+    if description:
+        data['description'] = description
+    if filter_object:
+        data['filter_object'] = filter_object
+
+    #TODO check that at least one has changed
+    result = current_session.query(Search).filter(
```
Comment (bobaekang) [inline amanuensis/resources/userdatamodel/userdatamodel_filterset.py:67]: This is a minor thing, but can we rename this variable to `filter_set`--sort of to match how the corresponding vars are named in, say, `create_filter_set()` and `delete_filter_set()`?

**Review by bobaekang — COMMENTED** [review]
I've added some comments. Please let me know if you need me to further clarify 😄

File: amanuensis/__init__.py (hunk 1 above)
Comment (dvenckusuchgo) [reply amanuensis/__init__.py:103]: In switching it for testing, I missed that one when I switched it back.  Corrected.

File: amanuensis/resources/userdatamodel/userdatamodel_filterset.py (hunk 3 above)
Comment (dvenckusuchgo) [reply amanuensis/resources/userdatamodel/userdatamodel_filterset.py:74]: No.  The update() method returns the number of records updated, not an id.  If you prefer an int, I can convert the id values before I return them.

File: amanuensis/resources/userdatamodel/userdatamodel_filterset.py (hunk 4 above)
Comment (dvenckusuchgo) [reply amanuensis/resources/userdatamodel/userdatamodel_filterset.py:67]: filter_object is the field name in the userPortalDataModel and it is a json object.  If it's minor, then it's best that we don't change it.

File: amanuensis/blueprints/filterset.py (hunk 2 above)
Comment (bobaekang) [reply amanuensis/blueprints/filterset.py]: Thank you for making the change.

File: amanuensis/resources/userdatamodel/userdatamodel_filterset.py (hunk 4 above)
Comment (bobaekang) [reply amanuensis/resources/userdatamodel/userdatamodel_filterset.py:67]: Ah, I think this is a miscommunication. I meant to suggest renaming `result` on line 67... 😅 

That said, based on your other comment, the `result` value is in fact not a filter set data but the number of updated records. In that case, can you rename `result` to something that better communicates that--maybe something like `num_updated` or `count_updated`?

File: amanuensis/resources/userdatamodel/userdatamodel_filterset.py (hunk 3 above)
Comment (bobaekang) [reply amanuensis/resources/userdatamodel/userdatamodel_filterset.py:74]: Thank you for clarifying what `result` is about. And yes, I'd appreciate if you can convert the id values in the response body to `int` type.

**Comment by dvenckusuchgo** [comment]
All recommended changes made.  Waiting for userPortalDataModel repo to have an updated release with the extend_data_request_model branch merged in.

**Review by dvenckusuchgo — COMMENTED** [review]
Changes made.  poetry dependency reference to userPortalDataModel bumped to 1.1.0.

**Comment by dvenckusuchgo** [comment]
Merging into pcdc_dev

**Pull request merged**